src/main/java/org/trigon/hardware/misc/leds/AddressableLEDStrip.java

```diff
@@ -0,0 +1,182 @@
+
+    @Override
+    void blink(Color firstColor, Color secondColor, double blinkingIntervalSeconds) {
+        double currentTime = Timer.getFPGATimestamp();
+        if (currentTime - lastLEDMovementTime > blinkingIntervalSeconds) {
+            lastLEDMovementTime = currentTime;
+            areLEDsOnForBlinking = !areLEDsOnForBlinking;
+        }
+        if (areLEDsOnForBlinking)
+            staticColor(firstColor);
+        else
+            staticColor(secondColor);
+    }
+
+    @Override
+    void staticColor(Color color) {
+        setLEDColors(color, 0, numberOfLEDs - 1);
+    }
+
+    @Override
+    void breathe(Color color, int breathingLEDs, double cycleTimeSeconds, boolean shouldLoop, boolean inverted, LarsonAnimation.BounceMode bounceMode) {
+        clearLEDColors();
+        inverted = this.inverted != inverted;
+        double moveLEDTimeSeconds = cycleTimeSeconds / numberOfLEDs;
+        double currentTime = Timer.getFPGATimestamp();
+        if (currentTime - lastLEDMovementTime > moveLEDTimeSeconds) {
+            lastLEDMovementTime = currentTime;
+            if (inverted)
+                lastBreatheLED--;
+            else
+                lastBreatheLED++;
+        }
+        if (inverted ? (lastBreatheLED < indexOffset) : (lastBreatheLED >= numberOfLEDs + indexOffset)) {
+            if (!shouldLoop) {
+                getDefaultCommand().schedule();
+                return;
+            }
+            lastBreatheLED = inverted ? indexOffset + numberOfLEDs : indexOffset;
+        }
+        for (int i = 0; i < breathingLEDs; i++) {
+            if (lastBreatheLED - i >= indexOffset && lastBreatheLED - i < indexOffset + numberOfLEDs)
+                LED_BUFFER.setLED(lastBreatheLED - i, color);
+            else if (lastBreatheLED - i < indexOffset + numberOfLEDs)
+                LED_BUFFER.setLED(lastBreatheLED - i + numberOfLEDs, color);
+        }
+    }
+
+    @Override
+    void colorFlow(Color color, double cycleTimeSeconds, boolean shouldLoop, boolean inverted) {
+        clearLEDColors();
+        inverted = this.inverted != inverted;
+        double moveLEDTimeSeconds = cycleTimeSeconds / numberOfLEDs;
+        double currentTime = Timer.getFPGATimestamp();
+        if (currentTime - lastLEDMovementTime > moveLEDTimeSeconds) {
+            lastLEDMovementTime = currentTime;
+            if (inverted)
+                amountOfColorFlowLEDs--;
+            else
+                amountOfColorFlowLEDs++;
+        }
+        if (inverted ? amountOfColorFlowLEDs < 0 : amountOfColorFlowLEDs >= numberOfLEDs) {
+            if (!shouldLoop) {
+                getDefaultCommand().schedule();
+                return;
+            }
+            amountOfColorFlowLEDs = inverted ? numberOfLEDs : 0;
+        }
+        if (inverted) {
+            setLEDColors(color, numberOfLEDs - amountOfColorFlowLEDs, numberOfLEDs - 1);
+            return;
+        }
+        setLEDColors(color, 0, amountOfColorFlowLEDs);
+    }
```

Comment:
Why do we need this? Isn't this just breathe? Also, not sure, but double check that this works properly and looks good. I think it might crash or look weird when it reaches the end

Reply:
I checked that it works